src/wasm.jl

```diff
@@ -54,16 +54,24 @@ end
 
 struct Block <: Instruction
   body::Vector{Instruction}
+
+  result::Union{WType, Void}
 end
+Block(body) = Block(body, Void())
```

Comment:
`Void()` -> nothing

Can you mention why this needs a change to the IR format? Would we need it even without the bytecode writer?

Reply:
To me it seems quite likely we'll need it at some point as you can't have an if block push a value to the stack without it. E.g.:
```
(module
  (func $this (result i64)
    i32.const 0
    if (result i64)
      i64.const 10
    else
      i64.const 12
    end
    return)
)
```
Won't compile through `wast2wasm` without the result mentioned in the if. I think it's reasonable an optimisation might look for a local being set at the end of each branch and restructuring to set the local after the if. Like a ternary operator. Currently the problem hasn't shown itself as blocks have been setting local variables and moving on without adding anything on the stack. (An if with no else can't have a result)

Currently the main reason though is that `base.wast` in the `arrays` branch does have results, and I'd like to be able to read it in with minimal manual reformatting and without building a bytecode ir.

Really whether it stays would depend on the goal of the internal format, which I'd assume is somewhere between coverage and strictly what's required by the compile pipeline.

Reply:
Yeah, that's fine if it maps directly to WASM features. The other option is to derive the type but storing it is probably a lot simpler. We should definitely be making use of this eventually, in order to make compiled code more compact.